### ISSUE TITLE
the Adam Optimizer

### ISSUE THREAD

**Issue by Dyfine** [Alibaba-MIIL/ImageNet21K#33]
Hi, thanks very much for sharing the code! I have read the code and found that the current Adam optimizer is not the AdamW (true weight decay) you mentioned in Appendix B.1 of your paper. I wonder do I get something wrong here?

**Comment by mrT23** [comment]
"true weight decay" is not adamw.
true weight decay means not doing weight decay on batch-norm and bias
https://github.com/Alibaba-MIIL/ImageNet21K/blob/main/src_files/helper_functions/general_helper_functions.py#L7

anyway, on CNNs adam and adamw give the same result. you will see a difference only on transformers.

**Comment by Dyfine** [comment]
I see. Thanks:)
